src/parser/literal.rs

```diff
@@ -39,25 +40,53 @@ pub fn literal(input: Span) -> IResult<Located<Literal>> {
     )(input)
 }
 
+/// Determines the radix of the next number in the input.
+///
+/// Tries to parse a number prefix if possible otherwise it assumes 10 for decimal numbers.
+fn number_radix(input: Span) -> IResult<u32> {
+    alt((
+        value(2, tag("0b")),
+        value(8, tag("0o")),
+        value(16, tag("0x")),
+        |i| Ok((i, 10)),
+    ))(input)
+}
+
 /// Parses a signed integer.
 ///
 /// This integer must be in the valid range for the `i64` type. If the number is outside this
 /// range, the parser will return an error.
 ///
 /// If the number is negative, there cannot be spaces between the minus sign and the digits of the
 /// number. That kind of expression will be parsed as an unary operation.
+///
+/// Numbers are accepted in decimal, binary, octal and hexadecimal notation.
+/// Each of these besides decimal require a prefix. These are:
+/// * binary `0b`
+/// * octal `0o`
+/// * hexadecimal `0x`
 fn number(input: Span) -> IResult<Located<i64>> {
     map_opt(
-        pair(opt(char('-')), digit1),
-        |(sign, digits_span): (Option<char>, Span)| {
-            let mut number = digits_span.fragment().parse::<i64>().ok()?;
+        tuple((opt(char('-')), number_radix, hex_digit1)),
```

Comment:
maybe you could "sandwich" this parser in `nom_locate::position` to avoid shifting the location manually:
```rust
tuple((position, opt(char('-')), number_radix, hex_digit1, position)), 
```
and then set the location to `Location::from(pos1) + Location::from(pos2)`.